proxmox/datadog_checks/proxmox/check.py

```diff
@@ -28,6 +30,93 @@ def check(self, _):
             self.set_metadata('version', version)
             self.gauge("api.up", 1, tags=self.base_tags)
 
-        except Exception as e:
-            self.log.error("Encountered an Exception when hitting the Proxmox API %s", e)
+        except (HTTPError, InvalidURL, ConnectionError, Timeout, JSONDecodeError) as e:
+            self.log.error(
+                "Encountered an Exception when hitting the Proxmox API %s: %s", self.config.proxmox_server, e
+            )
             self.gauge("api.up", 0, tags=self.base_tags)
+            return
+
+        resources_response = self.http.get(f"{self.config.proxmox_server}/cluster/resources")
+        resources_response_json = resources_response.json()
+        resources = resources_response_json.get("data", [])
+
+        external_tags = []
+
+        for resource in resources:
+            resource_type = resource.get('type')
+            node = resource.get('node')
+            resource_type_remapped = RESOURCE_TYPE_MAP.get(resource_type, resource_type)
+            resource_id = resource.get('id')
+            resource_name = resource.get('name')
+            if resource_name is None:
+                # some resource types don't have a name attribute
+                resource_name = resource.get(resource.get('type', ''))
+
+            hostname = None
+            if resource_type_remapped == VM_RESOURCE:
+                vmid = resource.get('vmid')
+                try:
+                    url = f"{self.config.proxmox_server}/nodes/{node}/qemu/{vmid}/agent/get-host-name"
+                    hostname_response = self.http.get(url)
+                    hostname_json = hostname_response.json()
+                except (HTTPError, InvalidURL, ConnectionError, Timeout, JSONDecodeError) as e:
+                    self.log.info(
+                        "Failed to get hostname for vm %s on node %s; endpoint: %s; %s",
+                        vmid,
+                        node,
+                        self.config.proxmox_server,
+                        e,
+                    )
+                    hostname_json = {}
+                hostname = hostname_json.get("data", {}).get("result", {}).get("host-name", resource_name)
+            elif resource_type_remapped == NODE_RESOURCE:
+                hostname = node
+
+            resource_tags = {
+                f'proxmox_type:{resource_type_remapped}',
+                f'proxmox_{resource_type_remapped}:{resource_name}',
```

Comment:
Are there any advantages to having a  `proxmox_{resource_type}` tag? 
I feel like having `promox_type` and `promox_name` would provide the same value, you can look up a resource by name, and filter by type, right?

Reply:
Ok I understand now, I'll use that approach!

Reply:
Updated!